Sources/FBLPromises/FBLPromise.m

```diff
@@ -202,6 +202,16 @@ - (NSMutableSet *__nullable)pendingObjects {
   }
 }
 
+- (void)addToPendingObjects:(id)object {
+  NSMutableSet *set = [self pendingObjects];
+
+  if (set) {
+    @synchronized(set) {
+      [set addObject:object];
```

Comment:
Can you please move the set creation logic over here from `pendingObjects` getter?

Reply:
sure